From: Michael Buesch <m@bues.ch>
Subject: [PATCH] slirp/smb: Move ncalrpc directory to tmp
Bug-Debian: http://bugs.debian.org/728876

The smbd forked by qemu still uses the default ncalrpc directory
in /var/run/samba. This may lead to problems, if /var/run/samba
does not exist (for example if /var/run is a tmpfs and the host
smbd was not started).

This leads to the following error message from samba
and an unworkable smbd:
	Failed to create pipe directory /var/run/samba/ncalrpc - No such file or directory

Fix this by pointing smbd to /tmp/qemu-smb.%d.%d/ncalrpc as ncalrpc directory.
Smbd will create the actual ncalrpc subdirectory on its own.

Signed-off-by: Michael Buesch <m@bues.ch>

---

Index: qemu/net/slirp.c
===================================================================
--- qemu.orig/net/slirp.c	2013-12-06 23:38:20.722329771 +0000
+++ qemu/net/slirp.c	2013-12-06 23:38:26.000000000 +0000
@@ -527,6 +527,7 @@
             "pid directory=%s\n"
             "lock directory=%s\n"
             "state directory=%s\n"
+            "ncalrpc dir=%s/ncalrpc\n"
             "log file=%s/log.smbd\n"
             "smb passwd file=%s/smbpasswd\n"
             "security = user\n"
@@ -539,6 +540,7 @@
             s->smb_dir,
             s->smb_dir,
             s->smb_dir,
+            s->smb_dir,
             s->smb_dir,
             s->smb_dir,
             s->smb_dir,
